index.js

```diff
@@ -1,6 +1,85 @@
-
 import { NativeModules } from 'react-native';
 
 const { RNSnowplowTracker } = NativeModules;
 
-export default RNSnowplowTracker;
+function _applyDefaults(input, defaults) {
```

Comment:
I think this function is directly modifying the object passed in by the user, and it feels like that breaks a contract.  We shouldn't modify objects that don't "belong" to us.

I think this function can be rewritten as
```
function _applyDefaults(input, defaults) {
  return {...defaults, ...input}
```
(untested!)

Reply:
Is it modifying the input? I understood it to copy the input to a new object and then return that?

I do like your way better though, much cleaner code. I'll give it a test.

Reply:
Good spot. Your way is much better IMO. Thanks for flagging!